msbuild/Xamarin.Mac.Tasks.Core/Tasks/Mmp.cs

```diff
@@ -0,0 +1,229 @@
+﻿//
+// MmpTask.cs
+//
+// Author:
+//   Aaron Bockover <abock@xamarin.com>
+//
+// Copyright 2014 Xamarin Inc.
+
+using System;
+using System.IO;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.MacDev.Tasks;
+using Xamarin.MacDev;
+
+namespace Xamarin.Mac.Tasks
+{
+	public class MmpTaskBase : ToolTask
+	{
+		protected override string ToolName {
+			get { return "mmp"; }
+		}
+
+		public string SessionId { get; set; }
+
+		[Required]
+		public string FrameworkRoot { get; set; }
+
+		[Required]
+		public string OutputPath { get; set; }
+
+		[Required]
+		public string ApplicationAssembly { get; set; }
+
+		[Required]
+		public string HttpClientHandler { get; set; }
+
+		[Required]
+		public string TargetFrameworkIdentifier { get; set; }
+
+		[Required]
+		public string TargetFrameworkVersion { get; set; }
+
+		[Required]
+		public string TLSProvider {	get; set; }
+
+		[Required]
+		public string SdkRoot {	get; set; }
+
+		[Required]
+		public ITaskItem AppManifest { get; set; }
+
+		[Required]
+		public string SdkVersion { get; set; }
+
+		public bool UseXamMacFullFramework { get; set; }
+
+		public string ApplicationName { get; set; }
+		public string Architecture { get; set; }
+		public string LinkMode { get; set; }
+		public bool Debug { get; set; }
+		public bool Profiling { get; set; }
+		public string I18n { get; set; }
+		public string ExtraArguments { get; set; }
+
+		public string [] ExplicitReferences { get; set; }
+		public string [] NativeReferences { get; set; }
+
+		public string IntermediateOutputPath { get; set; }
+		
+		protected override string GenerateFullPathToTool ()
+		{
+			return Path.Combine (FrameworkRoot, "bin", "mmp");
+		}
+
+		protected override bool ValidateParameters ()
+		{
+			XamMacArch arch;
+			return Enum.TryParse<XamMacArch> (Architecture, true, out arch);
+		}
+
+		protected override string GenerateCommandLineCommands ()
+		{
+			var args = new ProcessArgumentBuilder ();
+
+			args.Add ("/verbose");
+
+			if (Debug)
+				args.Add ("/debug");
+
+			if (!string.IsNullOrEmpty (OutputPath))
+				args.AddQuoted ("/output:" + Path.GetFullPath (OutputPath));
+
+			if (!string.IsNullOrEmpty (ApplicationName))
+				args.AddQuoted ("/name:" + ApplicationName);
+
+			if (TargetFrameworkIdentifier == "Xamarin.Mac")
+				args.Add ("/profile:Xamarin.Mac");
+			else if (TargetFrameworkVersion.StartsWith ("v", StringComparison.Ordinal))
+				args.Add ("/profile:" + TargetFrameworkVersion.Substring (1));
+
+			if (TargetFrameworkIdentifier == "Xamarin.Mac" || UseXamMacFullFramework) {
```

Comment:
Same question about the comparison. Also, could we just compare the target framework identifier once? Unless you have issues with increasing the depth of the if statement.